### PR TITLE
fix: close modal when clicking outside closely

### PR DESCRIPTION
**Pull request by mshaaban0**
### Expected behaviour
When clicking outside the modal content, the modal should close.

### Actual behaviour
When clicking close to the modal content edges, the modal doesn't close

https://user-images.githubusercontent.com/6163988/234340673-77ce3163-dfac-4f07-93aa-22177ce563bc.mov

## PR Checklist

- [x] I have read the relevant `readme.md` file(s)
- [x] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/main/packages/forma-36-react-components#commits)
- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Storybook stories are added/updated/not required
- [x] Usage notes are added/updated/not required
- [x] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [x] Doesn't contain any sensitive information

### CODE REVIEW

**Review by mattvanvoorst-contentful — APPROVED** [review]
Awesome thanks!

**Review by denkristoffer — COMMENTED** [review]
- How does this affect full height modals? 
- Changeset is missing

**Comment by mshaaban0** [comment]
> * How does this affect full height modals? 
> * Changeset is missing

@denkristoffer moved the padding to overlay as it does affect the full height when `allowHeightOverflow` is set to `true`

File: .changeset/cold-baboons-jog.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@contentful/f36-modal": patch
+---
+
+fix: close modal when clicking outside closely
```
Comment (denkristoffer) [inline .changeset/cold-baboons-jog.md:5]:
```suggestion
fix: always close modal when clicking outside
```

**Review by denkristoffer — APPROVED** [review]
✅

nit: I would change the wording for the changelog a bit.

**Pull request merged**